pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint_test.go

```diff
@@ -72,13 +80,100 @@ func TestAddEntrypoint(t *testing.T) {
 	}
 }
 
+type image struct {
+	config *v1.ConfigFile
+}
+
+// RawConfigFile implements partial.UncompressedImageCore
+func (i *image) RawConfigFile() ([]byte, error) {
+	return partial.RawConfigFile(i)
+}
+
+// ConfigFile implements v1.Image
+func (i *image) ConfigFile() (*v1.ConfigFile, error) {
+	return i.config, nil
+}
+
+// MediaType implements partial.UncompressedImageCore
+func (i *image) MediaType() (types.MediaType, error) {
+	return types.DockerManifestSchema2, nil
+}
+
+// LayerByDiffID implements partial.UncompressedImageCore
+func (i *image) LayerByDiffID(diffID v1.Hash) (partial.UncompressedLayer, error) {
+	return nil, fmt.Errorf("unknown diff_id: %v", diffID)
+}
+
+func mustRawManifest(t *testing.T, img v1.Image) []byte {
+	m, err := img.RawManifest()
+	if err != nil {
+		t.Fatalf("RawManifest() = %v", err)
+	}
+	return m
+}
+
+func mustRawConfigFile(t *testing.T, img v1.Image) []byte {
+	c, err := img.RawConfigFile()
+	if err != nil {
+		t.Fatalf("RawConfigFile() = %v", err)
+	}
+	return c
+}
+
+func getImage(t *testing.T, cfg *v1.ConfigFile) v1.Image {
+	rnd, err := partial.UncompressedToImage(&image{
+		config: cfg,
+	})
+	if err != nil {
+		t.Fatalf("random.Image() = %v", err)
+	}
+	return rnd
+}
+
+func mustConfigName(t *testing.T, img v1.Image) v1.Hash {
+	h, err := img.ConfigName()
+	if err != nil {
+		t.Fatalf("ConfigName() = %v", err)
+	}
+	return h
+}
+
 func TestGetRemoteEntrypoint(t *testing.T) {
-	ep, err := entrypoint.GetRemoteEntrypoint(entrypoint.NewCache(), kanikoImage)
+	img := getImage(t, &v1.ConfigFile{
+		ContainerConfig: v1.Config{
+			Entrypoint: []string{kanikoEntrypoint},
+		},
+	})
+	expectedRepo := "kaniko"
+	configPath := fmt.Sprintf("/v2/%s/blobs/%s", expectedRepo, mustConfigName(t, img))
+	manifestPath := fmt.Sprintf("/v2/%s/manifests/latest", expectedRepo)
+
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		switch r.URL.Path {
+		case "/v2/":
+			w.WriteHeader(http.StatusOK)
+		case configPath:
+			if r.Method != http.MethodGet {
+				t.Errorf("Method; got %v, want %v", r.Method, http.MethodGet)
+			}
+			w.Write(mustRawConfigFile(t, img))
+		case manifestPath:
+			if r.Method != http.MethodGet {
+				t.Errorf("Method; got %v, want %v", r.Method, http.MethodGet)
+			}
+			w.Write(mustRawManifest(t, img))
+		default:
+			t.Fatalf("Unexpected path: %v", r.URL.Path)
+		}
+	}))
+	defer server.Close()
+	image := path.Join(strings.TrimPrefix(server.URL, "http://"), "kaniko:latest")
+	ep, err := entrypoint.GetRemoteEntrypoint(entrypoint.NewCache(), image)
 	if err != nil {
 		t.Errorf("couldn't get entrypoint remote: %v", err)
 	}
 	if len(ep) != 1 {
-		t.Errorf("remote entrypoint should only have one item")
+		t.Fatalf("remote entrypoint should only have one item")
```

Comment:
Changed this to fatal as with `Errorf` it will panic on the next `if` :wink:

Reply:
Very nice. For brevity, you could check both at once:

```
if !reflect.DeepEqual(ep, []string{"expected-entrypoint"}) {
  t.Error(...)
}
```

Now that we're not tied to the Kaniko image you could also check that multiple-valued entrypoints work as expected.

Reply:
@ImJasonH good point :wink: